libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridSnapping.cpp

```diff
@@ -0,0 +1,239 @@
+        m_indexBoxUpperRight = m_lineEndIndex;
+
+        // The intermediate result in the subtraction cannot be less than zero.
+        m_smoothingRegionIndicator = CurvilinearGridNodeIndices(std::min<UInt>(1, m_lineEndIndex.m_n - m_lineStartIndex.m_n),
+                                                                std::min<UInt>(1, m_lineEndIndex.m_m - m_lineStartIndex.m_m));
+    }
+    else
+    {
+
+        if (m_points.size() == 3)
+        {
+            CurvilinearGridNodeIndices extentIndex = m_grid.GetNodeIndices(m_points[2]);
+
+            m_indexBoxLowerLeft = CurvilinearGridNodeIndices(std::min({m_lineStartIndex.m_m, m_lineEndIndex.m_m, extentIndex.m_m}),
+                                                             std::min({m_lineStartIndex.m_n, m_lineEndIndex.m_n, extentIndex.m_n}));
+            m_indexBoxUpperRight = CurvilinearGridNodeIndices(std::max({m_lineStartIndex.m_m, m_lineEndIndex.m_m, extentIndex.m_m}),
+                                                              std::max({m_lineStartIndex.m_n, m_lineEndIndex.m_n, extentIndex.m_n}));
+        }
+        else
+        {
+            auto [lowerExtentIndex, upperExtentIndex] = m_grid.ComputeBlockFromCornerPoints(m_points[2], m_points[3]);
+
+            m_indexBoxLowerLeft = CurvilinearGridNodeIndices(std::min(m_lineStartIndex.m_m, lowerExtentIndex.m_m), std::min(m_lineStartIndex.m_n, lowerExtentIndex.m_n));
+            m_indexBoxUpperRight = CurvilinearGridNodeIndices(std::max(m_lineEndIndex.m_m, upperExtentIndex.m_m), std::max(m_lineEndIndex.m_n, upperExtentIndex.m_n));
+        }
+    }
+
+    m_smoothingRegionIndicator = CurvilinearGridNodeIndices(std::min<UInt>(1, m_lineEndIndex.m_n - m_lineStartIndex.m_n),
+                                                            std::min<UInt>(1, m_lineEndIndex.m_m - m_lineStartIndex.m_m));
+}
+
+std::tuple<meshkernel::CurvilinearGridNodeIndices, meshkernel::CurvilinearGridNodeIndices>
+meshkernel::CurvilinearGridSnapping::ComputeLoopBounds(const CurvilinearGridNodeIndices& snappedNodeIndex) const
+{
+
+    if (m_points.size() == 2)
+    {
+        // Name from modgr4.f90
+        constexpr UInt nump = 80;
+        auto m1 = static_cast<UInt>(std::max<int>(1, snappedNodeIndex.m_m + 1 - nump * m_smoothingRegionIndicator.m_m) - 1);
+        auto m2 = static_cast<UInt>(std::min<int>(m_grid.m_numM, snappedNodeIndex.m_m + 1 + nump * m_smoothingRegionIndicator.m_m) - 1);
+        auto n1 = static_cast<UInt>(std::max<int>(1, snappedNodeIndex.m_n + 1 - nump * m_smoothingRegionIndicator.m_n) - 1);
+        auto n2 = static_cast<UInt>(std::min<int>(m_grid.m_numN, snappedNodeIndex.m_n + 1 + nump * m_smoothingRegionIndicator.m_n) - 1);
+        return {CurvilinearGridNodeIndices(m1, n1), CurvilinearGridNodeIndices(m2, n2)};
+    }
+    else
+    {
+        auto m1 = static_cast<UInt>(std::max<int>(m_indexBoxLowerLeft.m_m, snappedNodeIndex.m_m + 1 - 10000 * m_smoothingRegionIndicator.m_m - 1));
```

Comment:
Can we move 10000 as a static constant in the header file of CurvilinearGridSnapping?
Same applies to other numbers, such as nump, aspectRatio

Reply:
I have moved some of the magic numbers to constants defined in the header.
Its quite difficult to pick a good name for the 10000 and nump